### DISCUSSION TITLE
Support compiler flags  -Wno-error=name_of_warning . For example,  -Wno-error=deprecated-declarations  -Wno-error=macro-redefined -Wno-error=expansion-to-defined ...

### DISCUSSION THREAD

**Discussion by minskster** [ccache/ccache#1481] [Support]
Could you help to add support of -Wno-error=name_of_warning flag? I looked at "-fsyntax-only"  implementation and tried to add "-Wno-error=" in the same way because it's   I tried to implement it in the same way because this option from the same https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html chapter. 
Is there any other option that can be taken as the example ?

**Comment by jrosdahl** [comment]
Hi,

There should be no need to add special knowledge about `-Wno-error=...`. Could you describe what problem you are trying to solve?

**Reply by minskster** [reply]
Dear Joel,

I'm glad for your quick response :)  The following steps were tried:
1) Replace "-Wno-error" declaration to "-Wno-error="  in **src/ccache/compopt.cpp**
https://github.com/minskster/ccache/commit/c4d4ca964ea8108a4419843ec1b69e1ed0095ffd 
2) Add  "-Wno-error=" argument to  state.compiler_only_args  alike -fsyntax-only **src/ccache/argprocessing.cpp**
https://github.com/minskster/ccache/commit/48afecf10d77f321ee9e575e39304f658943d36c

The run attempt shows:
 cl : Command line error D8021: invalid numeric argument '/Wno-error=deprecated-declarations'

Could you give me the advice? The error has /Wno instead of -Wno . Maybe it's the problem.

**Reply by jrosdahl** [reply]
Hi @minskster,

You seem to describe a solution that you are trying out. However, it is still not clear what problem you are trying to fix with your solution. It is therefore hard to help out.

Please describe what problem you experience, not what you think the fix is. See https://github.com/ccache/ccache/blob/master/CONTRIBUTING.md#reporting-an-issue for some tips on information that is good to provide.

**Reply by jrosdahl** [reply]
> cl : Command line error D8021: invalid numeric argument '/Wno-error=deprecated-declarations'

It looks like `/Wno-error=deprecated-declarations` is passed to the MSVC compiler (cl.exe)? AFAICT, that is not a supported option for MSVC, so presumably the problem is not in ccache but in the build system you're using?

**Reply by minskster** [reply]
> > cl : Command line error D8021: invalid numeric argument '/Wno-error=deprecated-declarations'
> 
> It looks like `/Wno-error=deprecated-declarations` is passed to the MSVC compiler (cl.exe)? AFAICT, that is not a supported option for MSVC, so presumably the problem is not in ccache but in the build system you're using?

Dear Joel,  

I try to create a test project with only the existing problem because the real project is too complex. 

I have following build chain:
1)  cmake generates  Visual Studio 2017 15.0 solution 
2) a script patchs vcxproj files to work with ccache ( ) 
2.1) add the section `<PropertyGroup>` with:
2.1.1) `<CLToolExe>` where pointed path to ccache.exe renamed as cl.exe
2.1.2) `<TrackFileAccess>` false
2.1.3) `<UseMultiToolTask>` true
3)  For each `<ClCompile>` add `<DebugInformationFormat>OldStyle</DebugInformationFormat><ObjectFileName>$(IntDir)%(Filename).obj</ObjectFileName></ClCompile>`

And **ccache is cool ! It's really works** except the mentioned issue with some vcxproj . I cannot avoid the required options -Wno-error=... and some other that I have not mentioned yet because the example of the one implementation will be enough for me ;-)  

The project is built well without ccache redirect.  I have the issue only with some vcxproj files. The  options 
```
<ClCompile>
      <AdditionalIncludeDirectories> list of folders</AdditionalIncludeDirectories>
      <AdditionalOptions>%(AdditionalOptions) /Zc:__cplusplus -Wno-error=deprecated-declarations  -Wno-int-to-pointer-cast -Wno-error=
```

I started making a test demo project to find out the root of the issue.

Regards,
Vasily

**Reply by jrosdahl** [reply]
What is it that adds `<AdditionalOptions>%(AdditionalOptions) /Zc:__cplusplus -Wno-error=deprecated-declarations  -Wno-int-to-pointer-cast -Wno-error=` to the vcxproj file? Is it CMake (when generating the project) or the script you mention in step 2 or something else?

> I cannot avoid the required options -Wno-error=... 

This sounds strange to me since those are not MSVC (cl.exe) options. :thinking:

**Reply by minskster** [reply]
> What is it that adds `<AdditionalOptions>%(AdditionalOptions) /Zc:__cplusplus -Wno-error=deprecated-declarations -Wno-int-to-pointer-cast -Wno-error=` to the vcxproj file? Is it CMake (when generating the project) or the script you mention in step 2 or something else?
> 
> > I cannot avoid the required options -Wno-error=...
> 
> This sounds strange to me since those are not MSVC (cl.exe) options. 🤔

You are right! I've checked. Copilot says that the Microsoft Visual C++ (MSVC) compiler doesn’t directly support the -Wno-error= flag.  
-Wno-error=deprecated-declarations correspond to /wd4996 for MSVC. 

Now I'm trying to find out why these projects build without an error in non-ccache case. VS does not produce an error. Thank you for the showing me the way :)

**Reply by Martin-HZK** [reply]
@minskster Same issue, I have no idea why MSVC allows these incompatible warning messagesing inside the vcxproj files. How to avoid this when the problem is possibly caused by other thirdparty libraries.
